### ISSUE TITLE
Cannot read toString of null when using execSync

### ISSUE THREAD

**Issue by jordan-thoms** [shelljs/shelljs#415]
I've been occasionally getting the following error:

```
/private/var/folders/md/b4my_5v94kbcjzbdhs711fnr0000gn/T/shelljs_64464d02b4682598b994:4
  fs.writeFileSync('/var/folders/md/b4my_5v94kbcjzbdhs711fnr0000gn/T/shelljs_7bf01b37f53810b840b5', err ? err.code.toString() : '0');
                                                                                                                  ^

TypeError: Cannot read property 'toString' of null
    at /private/var/folders/md/b4my_5v94kbcjzbdhs711fnr0000gn/T/shelljs_64464d02b4682598b994:4:115
    at ChildProcess.exithandler (child_process.js:220:5)
    at emitTwo (events.js:87:13)
    at ChildProcess.emit (events.js:172:7)
    at maybeClose (internal/child_process.js:827:16)
    at Socket.<anonymous> (internal/child_process.js:319:11)
    at emitOne (events.js:77:13)
    at Socket.emit (events.js:169:7)
    at Pipe._onclose (net.js:475:12)
shell.js: internal error
Error: Command failed: "/usr/local/Cellar/node4-lts/4.4.0/bin/node" /var/folders/md/b4my_5v94kbcjzbdhs711fnr0000gn/T/shelljs_64464d02b4682598b994
    at checkExecSyncError (child_process.js:464:13)
    at Object.execSync (child_process.js:504:13)
    at execSync (/../shelljs/src/exec.js:81:11)

```

This seems to be caused by the code generated in https://github.com/shelljs/shelljs/blob/master/src/exec.js#L72 - it seems that node now returns an Error object when exec fails, which does not have a 'code' property.

This is with node v4.4.0 on OSX 10.11.3.

**Comment by nfischer** [comment]
Hmmm interesting. @ariporad would you be able to verify against OS X?

**Comment by ariporad** [comment]
I haven't been able to reproduce this. I tried both master and 0.6

**Comment by jordan-thoms** [comment]
This is still happening to me about 20% of the time when we run shelljs, particularly in cases where we execute many long-running commands using execSync. I think it may be triggered by the called program returning a non-zero error code, as it's when it access the err.code variable that it fails

**Comment by nfischer** [comment]
Could you give an example command please?

We have unit tests for this behavior specifically. Could you perhaps try running them on your system?

**Comment by nfischer** [comment]
Using [n\_shell](https://github.com/nfischer/n_shell) against the master branch of ShellJS:

```javascript
shelljs $ cat('long.sh');
sleep 10
exit 25

shelljs $ var ret = exec('bash long.sh');
undefined
shelljs $ ret.code
25
shelljs $ ret.code.toString()
'25'
```

This seems good to me, I don't see the issue. And I don't see why a bug would happen 20% of the time. Not sure if I can proceed further unless you can provide exactly what command you're running.

**Comment by nfischer** [comment]
@ariporad why is this blocked?

**Comment by ariporad** [comment]
Because we couldn't reproduce it, and were waiting for more info.

I personally would rather close it as [invalid], and it can be re-opened if needed.

**Comment by nfischer** [comment]
I think the "blocked" label may be counter intuitive here. When I see it, I assume it's blocked by another issue, not waiting for a response. Issues are almost always waiting for some sort of response, right?

Anyway, I'm fine with either closing and reopening or just leaving as is until we can reproduce the error.

**Comment by jordan-thoms** [comment]
Sorry, I hadn't seen the notifications on this. I haven't seen this error recently so we can close this and I'll reopen it and dig into this more if it becomes an issue again.

Thanks
